### PR TITLE
fix(#patch); aave v2 forks; Fix position issues

### PR DESCRIPTION
**Pull request by dmelotik**
Fixing positional issues we are experiencing in aave v2 fork subgraphs #1082 

testing: https://okgraph.xyz/?q=dmelotik%2Faave-v2-ethereum

Need to redeploy all forks after this is done.

### CODE REVIEW

File: subgraphs/aave-v2-forks/src/helpers.ts [hunk 1]
```diff
@@ -779,9 +779,10 @@ export function createAccount(accountID: string): Account {
   return account;
 }
 
-// returns the market based on the output token
-export function getMarketByOutputToken(
-  outputTokenID: string,
+// returns the market based on any auxillary token
+// ie, outputToken, vToken, or sToken
+export function getMarketByAuxillaryToken(
```
[the remark this answers is not in the document]
Comment (dmelotik) [reply subgraphs/aave-v2-forks/src/helpers.ts:784]: In theory this is unlimited, but is not a problem in any of the current protocols. Oneshot errors are caused by too high of a computation on an event?

File: subgraphs/aave-v2-forks/protocols/aave-v2/README.md [hunk 2]
```diff
@@ -59,6 +59,15 @@ Not applicable to Aave
 - The WETH gateways seem to be giving the most trouble on positional entities.
   - ie, a lot of the withdraw events not found is due to weth gateways acting as a third actor in transactions
 
+## Notes
+
+- Aave V2 Mainnet (not AMM), Polygon, and Avalanche have a collateral swap feature. This allows a user to switch collateral positions without doing a withdraw and deposit. The contract that is responsible for this is the Paraswap Router (a Paraswap router).
+  - Example transaction: https://snowtrace.io/tx/0x873b4019fa2d77dc9809c57b4c03d2db909d26776129225132c8e14f05c1319b
+  - e.g. aWAVAX is converted to WAVAX, WAVAX is swapped for USDC, USDC is converted to aUSDC then returned to the account executing the transaction
+  - There are also similar features with swapAndRepay routers and swapAndLiquidate routers.
+  - This does not affect any of the position data though.
```
[the remark this answers is not in the document]
Comment (dmelotik) [reply subgraphs/aave-v2-forks/protocols/aave-v2/README.md]: Cleared that up with the latest commit

File: subgraphs/aave-v2-forks/src/helpers.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaimehgb) [reply subgraphs/aave-v2-forks/src/helpers.ts:784]: From what I've seen, they happen when you loop through many entities. Every time you load an entity, it stays in memory at least until the event handler terminates. I think in both Euler Finance and Rocket Pool the issue appeared at around 100 entities. I imagine there should be way less most of the time?

**Review by jaimehgb — APPROVED** [review]

File: subgraphs/aave-v2-forks/src/helpers.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmelotik) [reply subgraphs/aave-v2-forks/src/helpers.ts:784]: In all the protocols there is less. Wouldn't be an issue unless we run into like an aave fork with a rari fuse like implementation

**Pull request merged**